runner_manager/clients/github.py

```diff
@@ -0,0 +1,134 @@
+"""
+
+GitHub Client class for the runner manager.
+
+Runner manager uses the GitHubkit library to interact with GitHub's API.
+As the library was generated with the OpenAPI spec of the free license,
+and the runner manager requires endpoints that are only available with
+the Enterprise Cloud license, we have to manually define the
+missing endpoints here.
+
+"""
+
+
+from functools import cached_property
+from typing import Dict, List, Optional
+
+from githubkit import GitHub as GitHubKit
+from githubkit.response import Response
+from githubkit.rest import RestNamespace as RestNamespaceKit
+from githubkit.rest.actions import ActionsClient as ActionsClientKit
+from githubkit.rest.models import GitHubRestModel
+from githubkit.utils import UNSET, Missing, exclude_unset
+
+
+class RunnerGroup(GitHubRestModel):
+    id: Optional[int] = None
```

Comment:
Are you sure you want to make it optional? It's not a problem at the moment, but it could potentially become one in the future if someone codes something using the `id`.

Reply:
I think we can't set it as required, as when we'll create the group, we have to provide a runnergroup object to the API call but we won't have the id of the group at this time. If we set it as required we will not be able to create this resource.